services/web/server/src/simcore_service_webserver/user_notifications.py

```diff
@@ -0,0 +1,87 @@
+from copy import deepcopy
+from datetime import datetime
+from enum import auto
+from typing import Any, Final
+from uuid import uuid4
+
+from models_library.users import UserID
+from models_library.utils.enums import StrAutoEnum
+from pydantic import BaseModel, NonNegativeInt, validator
+
+MAX_NOTIFICATIONS_FOR_USER_TO_SHOW: Final[NonNegativeInt] = 10
+MAX_NOTIFICATIONS_FOR_USER_TO_KEEP: Final[NonNegativeInt] = 100
+
+
+def get_notification_key(user_id: UserID) -> str:
+    return f"user_id={user_id}"
+
+
+class NotificationCategory(StrAutoEnum):
+    NEW_ORGANIZATION = auto()
+    STUDY_SHARED = auto()
+    TEMPLATE_SHARED = auto()
+
+
+class UserNotification(BaseModel):
+    id: str
```

Comment:
This was already rolled out with the data type being a string.
To make it a UUID, we'd require a **Redis data migration**. Which would go in a separate service. Since we don't have one and this is minor, I will keep it as a string for now. Will add some notes about it.